docs/non-normative/k8s-migration.md

```diff
@@ -0,0 +1,68 @@
+<!--- Hugo front matter used to generate the website version of this page:
+linkTitle: K8s migration
+--->
+
+# K8s semantic conventions stability migration
+
+Due to the significant number of modifications and the extensive user base
+affected by them, existing K8s instrumentations published by
+OpenTelemetry are required to implement a migration plan that will assist users in
+transitioning to the stable K8s semantic conventions.
+
+When existing K8s instrumentations published by OpenTelemetry are
+updated to the stable K8s semantic conventions, they:
+
+- SHOULD introduce an environment variable `OTEL_SEMCONV_STABILITY_OPT_IN` in
+  their existing major version, which accepts:
+  - `k8s` - emit the stable k8s conventions, and stop emitting
+    the old k8s conventions that the instrumentation emitted previously.
+  - `k8s/dup` - emit both the old and the stable k8s conventions,
+    allowing for a phased rollout of the stable semantic conventions.
+  - The default behavior (in the absence of one of these values) is to continue
+    emitting whatever version of the old k8s conventions the
+    instrumentation was emitting previously.
+- Need to maintain (security patching at a minimum) their existing major version
+  for at least six months after it starts emitting both sets of conventions.
+- May drop the environment variable in their next major version and emit only
+  the stable k8s conventions.
+
+Specifically for the Opentelemetry Collector:
+
+The transition will happen through two different feature gates.
+One for enabling the new schema called `semconv.k8s.enableStable`,
+and one for disabling the old schema called `semconv.k8s.disableLegacy`. Then:
+
+- On alpha the old schema is enabled by default (`semconv.k8s.disableLegacy` defaults to false),
+  while the new schema is disabled by default (`semconv.k8s.enableStable` defaults to false).
+- On beta/stable the old schema is disabled by default (`semconv.k8s.disableLegacy` defaults to true),
+  while the new is enabled by default (`semconv.k8s.enableStable` defaults to true).
+- It is an error to disable both schemas
+- Both schemas can be enabled with `--feature-gates=-semconv.disableLegacy,+semconv.k8s.enableStable`.
```

Comment:
Is there any way to make the gate names both "positive"? I think it will be clearer to users to do `--feature-gates=+semconv.k8s.enableLegacy,+semconv.k8s.enableStable` if you want both.

Reply:
Won't the `semconv.k8s.enableLegacy` default to false/disabled on `alpha` 🤔 ?

Reply:
It would have to start at beta, and then be moved to alpha following these [lifecycle steps](https://github.com/open-telemetry/opentelemetry-collector/tree/main/featuregate#feature-lifecycle). It wouldn't be the standard way for us to use the gate.

If we want to go from alpha -> beta -> stable we have to use the negative name.

Reply:
Ah I see now! I would be fine by this.

If we conclude with this approach it should be the same in the `hostmetrics` receiver and System SemConv as well (with the respective feature gates pair).
@mx-psi @dmitryax @braydonk @roger what do think?

Reply:
Thinking about this more following the standard feature-gate processor is probably better.  As long as we document how to emit both users will have a solution.